software/deploy/deploy.go

```diff
@@ -296,6 +240,74 @@ func getAirflowUILink(deploymentID string, deploymentURLs []houston.DeploymentUR
 	return ""
 }
 
+func getDeploymentIDForCurrentCommand(houstonClient houston.ClientInterface, wsID, deploymentID string, prompt bool) (string, []houston.Deployment, error) {
+	if wsID == "" {
+		return deploymentID, []houston.Deployment{}, ErrNoWorkspaceID
+	}
+
+	// Validate workspace
+	currentWorkspace, err := houston.Call(houstonClient.GetWorkspace)(wsID)
+	if err != nil {
+		return deploymentID, []houston.Deployment{}, err
+	}
+
+	// Get Deployments from workspace ID
+	request := houston.ListDeploymentsRequest{
+		WorkspaceID: currentWorkspace.ID,
+	}
+	deployments, err := houston.Call(houstonClient.ListDeployments)(request)
+	if err != nil {
+		return deploymentID, deployments, err
+	}
+
+	c, err := config.GetCurrentContext()
+	if err != nil {
+		return deploymentID, deployments, err
+	}
+
+	cloudDomain := c.Domain
```

Comment:
nit: You can just do this in the caller function, since you are anyways fetching the current context there, and pass the domain value to this function.